### PR TITLE
RN-783 [fix]: Fix data element names not being consistent when pulling from DHIS2

### PR DESCRIPTION
**Pull request by rohan-bes**
### Issue RN-783:
Issues: 8, 9, 14, 16

Reworked fetchDataElementMetadata in DhisService to pull category option combos and combine with data element name. This is the behaviour of pulling data element metadata with fetchAnalytics, so keeping it consistent. Might have to do some regression on the `tableOfEvents` and `tableOfDataValues` legacy reports, but they seemed to still work fine when I took a look at them.

### CODE REVIEW

File: packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@
  */
 import { Request, NextFunction, Response } from 'express';
 import { PermissionsError } from '@tupaia/utils';
+import { ajvValidate } from '@tupaia/tsutils';
```
Comment (rohan-bes) [inline packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts:7]: Unrelated change, just linting warning

File: packages/entity-server/src/routes/hierarchy/format/formatEntityForResponse.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
-import { reduceToDictionary, reduceToArrayDictionary } from '@tupaia/utils';
```
Comment (rohan-bes) [inline packages/entity-server/src/routes/hierarchy/format/formatEntityForResponse.ts:6]: Unrelated cleanup

File: packages/data-broker/src/services/dhis/translators/DhisTranslator.ts [hunk 3]
```diff
@@ -221,10 +221,19 @@ export class DhisTranslator {
...
-          ...metadataGroupedByCode[dataElementCode],
-          code,
-        }));
+        .map(({ code, dataElementCode, config }) => {
+          const { id, name } = metadataGroupedByCode[dataElementCode];
+          const { categoryOptionCombo: categoryOptionComboCode } = config;
+          const categoryOptionCombo = categoryOptionCombos.find(
+            item => item.code === categoryOptionComboCode,
+          );
+          const dataElementName = formatInboundDataElementName(name, categoryOptionCombo?.name);
```
Comment (rohan-bes) [inline packages/data-broker/src/services/dhis/translators/DhisTranslator.ts:230]: Now the formatting for the data element name matches what comes back with the analytics `metadata` object.

**Review by chris-pollard — APPROVED** [review]
LGTM!
